modules/__tests__/TestSequences/EncodedReservedCharacters.js

```diff
@@ -0,0 +1,46 @@
+import expect from 'expect'
+import execSteps from './execSteps'
+
+export default (history, done) => {
+  const steps = [
+    () => {
+      // encoded string
+      const pathname = '/view/%23abc'
+      history.replace(pathname)
+    },
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/%23abc'
+      })
+
+      // encoded object
+      const pathname = '/view/%23abc'
+      history.replace({ pathname })
+    },
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/%23abc'
+      })
+      // unencoded string
+      const pathname = '/view/#abc'
+      history.replace(pathname)
+    }
+    ,
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/',
+        hash: '#abc'
+      })
+      // unencoded object
+      const pathname = '/view/#abc'
+      history.replace({ pathname })
+    },
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/#abc'
```

Comment:
I'm not actually sure about this test case. It passes, but I'm not sure if the behavior is desirable. This only happens if the user passes a bad location object.